From a7732e01ed528414d58200c9a1d7236279af48e1 Mon Sep 17 00:00:00 2001
From: Andrejs Cainikovs <andrejs.cainikovs@toradex.com>
Date: Fri, 26 Aug 2022 13:27:00 +0200
Subject: [PATCH 215/233] [FIO tonxp] arm64: dts: imx8mp: add rdc config

Suppress warning by adding default RDC configuration.

Upstream-Status: Inappropriate [other]
- NXP downstream specific

Signed-off-by: Andrejs Cainikovs <andrejs.cainikovs@toradex.com>
(cherry picked from commit 16de213af305ce987d5688429c4b63cc57c48841)
Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 arch/arm/dts/imx8mp-verdin-u-boot.dtsi | 45 ++++++++++++++++++++++++++
 1 file changed, 45 insertions(+)

diff --git a/arch/arm/dts/imx8mp-verdin-u-boot.dtsi b/arch/arm/dts/imx8mp-verdin-u-boot.dtsi
index a57ad45ed63..b56deded32a 100644
--- a/arch/arm/dts/imx8mp-verdin-u-boot.dtsi
+++ b/arch/arm/dts/imx8mp-verdin-u-boot.dtsi
@@ -3,6 +3,7 @@
  * Copyright 2022 Toradex
  */
 
+#include "imx8mp-sec-def.h"
 #include "imx8mp-u-boot.dtsi"
 
 / {
@@ -18,6 +19,50 @@
 		u-boot,dm-spl;
 		wdt = <&wdog1>;
 	};
+
+	mcu_rdc {
+		compatible = "imx8m,mcu_rdc";
+			    /* rdc config when MCU starts
+			     * master
+			     *   SDMA3p --> domain 1
+			     *   SDMA3b --> domain 1
+			     *   SDMA3_SPBA2 --> domain 1
+			     * peripheral:
+			     *   SAI3   --> Only domain 1 can access
+			     *   UART4  --> Only domain 1 can access
+			     *   GPT1   --> Only domain 1 can access
+			     *   SDMA3  --> Only domain 1 can access
+			     *   I2C3   --> Only domain 1 can access
+			     * memory:
+			     *   TCM    --> Only domain 1 can access (0x7E0000~0x81FFFF)
+			     *   DDR    --> Only domain 1 can access (0x80000000~0x81000000)
+			     * end.
+			     */
+		start-config = <
+			    RDC_MDA  RDC_MDA_SDMA3p DID1 0x0 0x0
+			    RDC_MDA  RDC_MDA_SDMA3b DID1 0x0 0x0
+			    RDC_MDA  RDC_MDA_SDMA3_SPBA2 DID1 0x0 0x0
+			    RDC_PDAP RDC_PDAP_SAI3  PDAP_D1_ACCESS 0x0 0x0
+			    RDC_PDAP RDC_PDAP_UART4 PDAP_D1_ACCESS 0x0 0x0
+			    RDC_PDAP RDC_PDAP_GPT1  PDAP_D1_ACCESS 0x0 0x0
+			    RDC_PDAP RDC_PDAP_SDMA3 PDAP_D1_ACCESS 0x0 0x0
+			    RDC_PDAP RDC_PDAP_I2C3  PDAP_D1_ACCESS 0x0 0x0
+			    RDC_MEM_REGION 22 TCM_START TCM_END MEM_D1_ACCESS
+			    RDC_MEM_REGION 39 M4_DDR_START M4_DDR_END MEM_D1_ACCESS
+			    0x0      0x0            0x0  0x0 0x0
+			  >;
+			    /* rdc config when MCU stops
+			     * memory:
+			     *   TCM    --> domain 0/1 can access (0x7E0000~0x81FFFF)
+			     *   DDR    --> domain 0/1 can access (0x80000000~0x81000000)
+			     * end.
+			     */
+		stop-config = <
+			    RDC_MEM_REGION 22 TCM_START TCM_END MEM_D0D1_ACCESS
+			    RDC_MEM_REGION 39 M4_DDR_START M4_DDR_END MEM_D0D1_ACCESS
+			    0x0      0x0            0x0  0x0 0x0
+			  >;
+	};
 };
 
 &clk {
-- 
2.40.1

